lib/snapshot.js

```diff
@@ -392,16 +394,24 @@ function check_site_queue() {
 
 			var s_details = site_queue[0];
 			site_queue.shift();
-			logger.debug( process.pid + ': snapping: ' + s_details.url );
-			take_snapshot(
-				s_details.url,
-				s_details.file,
-				s_details.width,
-				s_details.height,
-				s_details.screenWidth,
-				s_details.screenHeight,
-				s_details.scaleFactor
-			);
+
+			if( s_details.action === 'analyze' ) {
+				logger.debug(process.pid + ': analyzing: ' + s_details.url);
+				analyze_url.analyze( s_details.url, s_details.key );
+				time_downloading = 0;
+				downloading = false;
```

Comment:
I think it would be better to only set `downloading = true` in the else condition below. Then you could remove this statement and simplify the code a little.